src/telemetry/metrics/tls_config_reload.rs

```diff
@@ -25,41 +47,61 @@ enum Status {
     InvalidTrustAnchors,
     InvalidPrivateKey,
     InvalidEndEntityCert,
-    Io { path: PathBuf, errno: Option<Errno>, },
+    Io { path: PathBuf, errno: Option<Errno> },
 }
 
-// ===== impl TlsConfigReload =====
+// ===== impl Sensor =====
 
-impl TlsConfigReload {
-    pub fn success(&mut self, when: &SystemTime) {
-        let t = when
+impl Sensor {
+    pub fn reloaded(&mut self) {
+        let t = SystemTime::now()
             .duration_since(UNIX_EPOCH)
             .expect("times must be after UNIX epoch")
             .as_secs();
-        self.last_reload = Some(t.into());
 
-        self.by_status.scopes
-            .entry(Status::Reloaded)
-            .or_insert_with(|| Counter::default())
-            .incr()
+        if let Ok(mut inner) = self.0.lock() {
+            inner.last_reload = Some(t.into());
+
+            inner
+                .by_status
+                .scopes
+                .entry(Status::Reloaded)
+                .or_insert_with(|| Counter::default())
+                .incr();
+        }
     }
 
-    pub fn error(&mut self, e: tls::ConfigError) {
-        self.by_status.scopes
-            .entry(e.into())
-            .or_insert_with(|| Counter::default())
-            .incr()
+    pub fn failed(&mut self, e: tls::ConfigError) {
+        if let Ok(mut inner) = self.0.lock() {
+            inner
+                .by_status
+                .scopes
+                .entry(e.into())
+                .or_insert_with(|| Counter::default())
+                .incr();
+        }
     }
 }
 
-impl fmt::Display for TlsConfigReload {
+// ===== impl Fmt =====
+
+impl fmt::Display for Fmt {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        if !self.by_status.scopes.is_empty() {
+        let lock = match self.0.upgrade() {
```

Comment:
TIOLI: 
```rust
let lock = match self.0
     .upgrade()
     .and_then(|lock| lock.lock().ok())
{
    None => return Ok(()),
    Some(inner) => inner
}
```

Reply:
i'm pretty sure this doesn't work because `lock's` lifetime is too short

Reply:
Oh, yeah, that's probably the case. Nevermind.